docs/architecture/adr/0004-diagrams-first-documentation.md

```
@@ -0,0 +1,158 @@
+# ADR-0005: Diagrams-First Documentation
+
+## Status
```

markdownlint-cli2 (0.17.2)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)